server/src/main/java/server/haengdong/application/BillActionService.java

```diff
@@ -0,0 +1,45 @@
+package server.haengdong.application;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import server.haengdong.application.request.BillActionAppRequest;
+import server.haengdong.domain.Action;
+import server.haengdong.domain.BillAction;
+import server.haengdong.domain.Event;
+import server.haengdong.persistence.ActionRepository;
+import server.haengdong.persistence.BillActionRepository;
+import server.haengdong.persistence.EventRepository;
+
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class BillActionService {
+
+    private final BillActionRepository billActionRepository;
+    private final ActionRepository actionRepository;
+    private final EventRepository eventRepository;
+
+    @Transactional
+    public void saveAllBillAction(String eventToken, List<BillActionAppRequest> requests) {
+        Event event = eventRepository.findByToken(eventToken)
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 이벤트 토큰입니다."));
+        long lastSequence = getLastSequence(event);
+
+        List<BillAction> billActions = new ArrayList<>();
+        for (BillActionAppRequest request : requests) {
+            Action action = new Action(event, ++lastSequence);
+            BillAction billAction = request.toBillAction(action);
+            billActions.add(billAction);
+        }
+        billActionRepository.saveAll(billActions);
+    }
+
+    private long getLastSequence(Event event) {
+        return actionRepository.findLastByEvent(event)
+                .map(Action::getSequence)
+                .orElse(0L);
+    }
```

Comment:
```suggestion
        Action action = createStartAction(event);

        for (BillActionAppRequest request : requests) {
            BillAction billAction = request.toBillAction(action);
            billActionRepository.save(billAction);
            action = action.next();
        }
    }

    private Action createStartAction(Event event) {
        return actionRepository.findLastByEvent(event)
                .map(Action::next)
                .orElse(Action.createFirst(event));
    }
```

Action의 sequence가 Service 층으로 노출하는 것이 불필요하고 sequence의 시작점이 1, 다음 순서가 +1 이라는 로직은 Action의 역할로 판단하여 수정했습니다.

Reply:
좋다고 생각합니다. #47 도 동일한 구조로 수정했습니다.